### ISSUE TITLE
JSONDecodeError in test_cli_lifecycle

### ISSUE THREAD

**Issue by cygnusv** [nucypher/nucypher#1463]
There are some tests that sometimes, at random, fail due to a `JSONDecodeError`. The problem is that that piece of code is expacting a JSON string which is the output of a json-rpc call, but it's also getting a stdout output like if it were using the `--debug` flag.

Example: https://circleci.com/gh/nucypher/nucypher/56027#tests/containers/5

```
self = <json.decoder.JSONDecoder object at 0x7f9060df2cd0>
s = 'Unlocking keyring.\nFinished unlocking.\nStarting Learning Loop.\nFinished learning about all seednodes.\n\n\n    / \...-11-19_11-23-05.009732/known_nodes/metadata\nOK | Responded to IPC request #1574162584 with None bytes, took 0:00:40\n'
idx = 0

    def raw_decode(self, s, idx=0):
        """Decode a JSON document from ``s`` (a ``str`` beginning with
        a JSON document) and return a 2-tuple of the Python
        representation and the index in ``s`` where the document ended.
    
        This can be used to decode a JSON document from a string that may
        have extraneous data at the end.
    
        """
        try:
            obj, end = self.scan_once(s, idx)
        except StopIteration as err:
>           raise JSONDecodeError("Expecting value", s, err.value) from None
E           json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

I have no idea why is this happening. Perhaps some misconfiguration of our logging (and its associated cache) during tests.

**Comment by michwill** [comment]
Perhaps some logging is going to stdout instead of stderr?

**Comment by cygnusv** [comment]
BTW, cannot reproduce locally. So far, I only encounter this on CircleCI.

**Comment by KPrasch** [comment]
Digging a bit deeper in the stack trace there is a `Unhandled error during node learning.`

I am able to reproduce this by running this group of tests: `pytest tests/cli/ursula/ tests/cli/test_cli_lifecycle.py `

```
Unhandled error during node learning: Traceback (most recent call last):
  File "/home/k/.local/share/virtualenvs/nucypher-iNjwDRsw/lib/python3.6/site-packages/twisted/internet/base.py", line 1283, in run
    self.mainLoop()
  File "/home/k/.local/share/virtualenvs/nucypher-iNjwDRsw/lib/python3.6/site-packages/twisted/internet/base.py", line 1292, in mainLoop
    self.runUntilCurrent()
  File "/home/k/.local/share/virtualenvs/nucypher-iNjwDRsw/lib/python3.6/site-packages/twisted/internet/base.py", line 913, in runUntilCurrent
    call.func(*call.args, **call.kw)
  File "/home/k/.local/share/virtualenvs/nucypher-iNjwDRsw/lib/python3.6/site-packages/twisted/internet/task.py", line 239, in __call__
    d = defer.maybeDeferred(self.f, *self.a, **self.kw)
--- <exception caught here> ---
  File "/home/k/.local/share/virtualenvs/nucypher-iNjwDRsw/lib/python3.6/site-packages/twisted/internet/defer.py", line 151, in maybeDeferred
    result = f(*args, **kw)
  File "/home/k/Git/nucypher/nucypher/network/nodes.py", line 565, in keep_learning_about_nodes
    self.learn_from_teacher_node(eager=False)
  File "/home/k/Git/nucypher/nucypher/network/nodes.py", line 716, in learn_from_teacher_node
    fleet_checksum=self.known_nodes.checksum)
  File "/home/k/Git/nucypher/nucypher/network/middleware.py", line 226, in get_nodes_via_rest
    payload = bytes().join(bytes(VariableLengthBytestring(n)) for n in announce_nodes)
  File "/home/k/Git/nucypher/nucypher/network/middleware.py", line 226, in <genexpr>
    payload = bytes().join(bytes(VariableLengthBytestring(n)) for n in announce_nodes)
  File "/home/k/.local/share/virtualenvs/nucypher-iNjwDRsw/lib/python3.6/site-packages/bytestring_splitter/__init__.py", line 262, in __init__
    self.message_as_bytes = bytes(message)
  File "/home/k/Git/nucypher/nucypher/characters/lawful.py", line 970, in __bytes__
    interface_info = VariableLengthBytestring(bytes(self.rest_interface))
  File "/home/k/Git/nucypher/nucypher/characters/lawful.py", line 957, in rest_interface
    return self.rest_server.rest_interface
builtins.AttributeError: 'Ursula' object has no attribute 'rest_server'


```

**Comment by KPrasch** [comment]
Turns out the `AttributeError` is a separate Issue altogether - Move here <https://github.com/nucypher/nucypher/issues/1469>

**Comment by KPrasch** [comment]
Okay the `JSONDecodeError` is happening because of an Emitter issue, wherein non-RPC string data is written to standard out despite alice passing the `--json-ipc` flag  (The banner and keyring logging messages are echoed).   Unsure what the solution is yet.

**Comment by michwill** [comment]
Can the log emitter just always write to stderr?

**Comment by KPrasch** [comment]
Fixed in fede665
